### PR TITLE
Feature/easy computer

### PR DESCRIPTION
**Pull request by grace000**
This PR addresses this [user story](https://trello.com/c/H1sXGdaO/26-computer-player-beatability-easy).

New Features:
- User can select between a game with two humans or one human/one computer
intro

- User can play against a computer player that selects random moves
easy-computer

### CODE REVIEW

File: assets/js/pages/game.js [hunk 1]
```diff
@@ -43,62 +46,65 @@ postNewMark(requestedMove) {
...
-      let currentPlayer = data.current_player;
+      const data = response.data;
+      const boardData = data.board;
+      const gameStatus = data.game_status;
+      const currentPlayer = data.current_player;
       this.setState({
           gameStatus: gameStatus,
           moves: Object.assign(this.state.moves, boardData),
           currentPlayer: currentPlayer,
           message: currentPlayer == "X" ? "Player one, please select a spot on the board!" : "Player two, please select a spot on the board!"
```
Comment (bgerstle) [inline assets/js/pages/game.js]: [JS template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) are pretty nifty, can reduce duplicate boilerplate here.

File: assets/js/pages/game.js [hunk 2]
```diff
@@ -11,22 +10,26 @@ class Game extends Component {
...
+        const currentPlayer = data.current_player;
+        const gameStatus = data.game_status;
+        const gameType = data.game_type;
         this.setState({
             gameStatus: gameStatus,
             moves: Object.assign(this.state.moves, boardData),
             currentPlayer: currentPlayer,
+            gameType: gameType
         })
     }).catch(error => console.log(error))
```
Comment (hnlee) [inline assets/js/pages/game.js]: Probably want a better way of handling this error than leaving as `console.log`.

File: assets/js/pages/game.js [hunk 3]
```diff
@@ -43,62 +46,65 @@ postNewMark(requestedMove) {
...
         }
     }).then((response) => {
-      let data = response.data;
-      let boardData = data.board;
-      let gameStatus = data.game_status;
-      let currentPlayer = data.current_player;
+      const data = response.data;
+      const boardData = data.board;
+      const gameStatus = data.game_status;
+      const currentPlayer = data.current_player;
```
Comment (hnlee) [inline assets/js/pages/game.js]: There's a lot of duplication between here and the call in the `componentDidMount` method above. Maybe worth extracting out into method that takes a `response` object?

File: assets/js/pages/game.js [hunk 4]
```diff
@@ -43,62 +46,65 @@ postNewMark(requestedMove) {
...
 
+removeUnderscores(string) {
+  const newString = string.replace(/_/g, " ");
+  return newString
+}
+
 removeErrorMessage() {
     this.setState({error: " "});
 }
   render() {
```
Comment (hnlee) [inline assets/js/pages/game.js:73]: This is indented one level too deep.

File: assets/js/pages/game.js [hunk 5]
```diff
@@ -43,62 +46,65 @@ postNewMark(requestedMove) {
...
       if(errorCode == 400) {
         this.setState({error: errorMessage});
       }
     });
 }
 
+removeUnderscores(string) {
+  const newString = string.replace(/_/g, " ");
+  return newString
+}
```
Comment (hnlee) [inline assets/js/pages/game.js]: Is this really the responsibility of the `Game` component? Feels more like a helper/utils function.

File: assets/js/pages/game.test.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import * as React from 'react'
```
Comment (hnlee) [inline assets/js/pages/game.test.js:1]: A little confused about why the tests are not in their own directory inside `assets`...sorry, I probably should have taken a closer look before, but I didn't notice until now. Do they not run if you don't have it structured this way?

File: assets/js/pages/game.js (hunk 3 above)
Comment (ssunday) [reply assets/js/pages/game.js]: +1

File: assets/js/pages/game.js [hunk 7]
```diff
@@ -43,62 +46,65 @@ postNewMark(requestedMove) {
...
-          <section className="phx-hero">
-            <h1>Welcome to Tic Tac Toe!</h1>
-          </section>
-          <section> 
-            { (this.state.gameStatus == "winner" || this.state.gameStatus == "draw") ? 
+        <section className="phx-hero">
+          <h1>{this.removeUnderscores(this.props.location.state.message)} game</h1>
+        </section>
+        <section> 
+          { (this.state.gameStatus == "winner" || this.state.gameStatus == "draw") ?
```
Comment (ssunday) [inline assets/js/pages/game.js]: Could you pull this out into a function to test if the game is over?

File: assets/js/pages/game.test.js (hunk 6 above)
Comment (ssunday) [reply assets/js/pages/game.test.js:1]: I've seen this structure on a few projects, and it `works`, but IMO it's really not a good structure so I'd rec moving it

File: lib/tic_tac_toe/game.ex [hunk 8]
```diff
@@ -3,23 +3,26 @@ defmodule TicTacToe.Game do
...
+        %Game {
+            game_type: game_type,
+            game_status: "", 
+            current_player: "X", 
+            incoming_move: nil,
+            board: Board.empty_board
         }
     end
-
+
```
Comment (ssunday) [inline lib/tic_tac_toe/game.ex:25]: Extra spacing

File: lib/tic_tac_toe_web/controllers/game_controller.ex [hunk 9]
```diff
@@ -50,18 +59,47 @@ defmodule TicTacToeWeb.GameController do
...
+        |> json_to_map
+    end
+
+    def create_move(connection, params) do
+        mapped_data = mapped_request(connection, params)
+
+        case mapped_data.game_type do
+            "human_vs_human" ->
+                create_human_move(connection, params)
+            "easy_computer" ->
```
Comment (ssunday) [inline lib/tic_tac_toe_web/controllers/game_controller.ex:73]: The game type is a symbol one place and a string another...could it be made consistent or turned into constants?

File: lib/tic_tac_toe_web/router.ex [hunk 10]
```diff
@@ -17,7 +17,8 @@ defmodule TicTacToeWeb.Router do
     pipe_through :browser
 
     get "/", PageController, :index
-    get "/new_game", GameController, :new_game
+    get "/new_game", PageController, :index
```
Comment (ssunday) [inline lib/tic_tac_toe_web/router.ex:20]: Why is new game in the page controller instead of game controller?

**Pull request merged**